### PR TITLE
Remove all llvm_asm macros

### PR DESCRIPTION
**Pull request by mbuesch**
Remove all `llvm_asm!` macros and require a recent compiler with `asm!` support.
As discussed in #104

### CODE REVIEW

**Comment by mbuesch** [comment]
Update: Removed the compile_error, because it doesn't work. The compile aborts at the earlier feature request.

**Comment by mbuesch** [comment]
CI fails due to ancient compiler `nightly-2021-01-07`, as expected.

**Comment by mbuesch** [comment]
Update: Remove now unnecessary build dependency `rustversion`.

File: src/lib.rs [hunk 1]
```diff
@@ -131,8 +131,7 @@
 //! * To enable the crate's runtime environment, use the `rt` feature.
 
 #![no_std]
-#![cfg_attr(avr_device_asm_macro, feature(asm_experimental_arch))]
-#![cfg_attr(not(avr_device_asm_macro), feature(llvm_asm))]
+#![feature(asm_experimental_arch)] // For asm! macro.
```
Comment (Rahix) [inline src/lib.rs:134]: Hmm, I feel like we should only enable this when building for avr, so something like

```rust
#![cfg_attr(target_arch = "avr", feature(asm_experimental_arch))]
```

**Review by Rahix — COMMENTED** [review]
Hey, thanks for taking care of this!  I have one minor nit, everything else looks great!

**Comment by Rahix** [comment]
Ah, and CI is failing because it is still using the old nightly.  I'll push a commit to upgrade it, then you can rebase to (hopefully) get it to pass.

**Comment by Rahix** [comment]
Done in commit 2e8d033ae636 ("ci: Upgrade to a new compiler version").  Please rebase when you find the time :)

**Comment by mbuesch** [comment]
Thanks. I pushed an updated branch.

**Review by Rahix — APPROVED** [review]
Thanks!

**Pull request merged**